### PR TITLE
feature / drag and drop functionality for payment methods

### PR DESCRIPTION
**Pull request by tomas862**

### CODE REVIEW

**Comment by tomas862** [comment]
- [ ] missing when page loaded input prefilling with positions and sorting of payment methods

- [ ] missing position submit functionality

- [ ] missing sorting payment options both ps 1.6 and 1.7 in front-office

File: src/Repository/PaymentMethodRepository.php [hunk 1]
```diff
@@ -39,11 +39,23 @@
...
+use PDOStatement;
 use PrestaShopDatabaseException;
 use PrestaShopException;
 
-class PaymentMethodRepository
+/**
+ * @deprecated - outside code must always use interface. Use PaymentMethodRepositoryInterface instead.
+ * In Containers use PaymentMethodRepositoryInterface::class
+ */
+class PaymentMethodRepository extends  AbstractRepository implements PaymentMethodRepositoryInterface
```
Comment (margud) [inline src/Repository/PaymentMethodRepository.php:52]: after extend there are 2 spaces :D

File: src/Service/SettingsSaveService.php [hunk 2]
```diff
@@ -79,21 +80,24 @@ class SettingsSaveService
      * @var MolCarrierInformationService
      */
     private $carrierInformationService;
+    private $paymentMethodPositionHandler;
```
Comment (margud) [inline src/Service/SettingsSaveService.php:83]: type hint missing

File: src/Service/SettingsSaveService.php [hunk 3]
```diff
@@ -111,6 +115,10 @@ public function saveSettings(&$errors = [])
         $mollieApiKey = Tools::getValue(Config::MOLLIE_API_KEY);
         $mollieApiKeyTest = Tools::getValue(Config::MOLLIE_API_KEY_TEST);
         $mollieProfileId = Tools::getValue(Config::MOLLIE_PROFILE_ID);
+        $paymentOptionPositions = Tools::getValue('payment_option_position');
```
Comment (margud) [inline src/Service/SettingsSaveService.php:118]: why dont you create const for this?

**Review by margud — APPROVED** [review]

File: src/Repository/PaymentMethodRepository.php (hunk 1 above)
Comment (tomas862) [reply src/Repository/PaymentMethodRepository.php:52]: wait for me once I review your code :D

**Pull request merged**